### ISSUE TITLE
Keras optimizer uses non-trainable variables instead of constants for hyperparameters

### ISSUE THREAD

**Issue by behzad-a** [tensorflow/tensorflow#45558]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): No
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux CentOS 7.8
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: N/A
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): 2.2.0
- Python version: 3.7.4
- Bazel version (if compiling from source): N/A
- GCC/Compiler version (if compiling from source): N/A
- CUDA/cuDNN version: N/A
- GPU model and memory: N/A

**Describe the current behavior**

Keras OptimizerV2 converts hyperparameters which are python data types to non-trainable TF Variables. Is there a reason they are not converted to tf.constant instead? Here's the relevant piece of code from `optimizer_v2.py`:

```python
def _create_hypers(self):
    if self._hypers_created:
      return
    # Iterate hyper values deterministically.
    for name, value in sorted(self._hyper.items()):
      if isinstance(
          value, (ops.Tensor, tf_variables.Variable)) or callable(value):
        continue
      else:
        self._hyper[name] = self.add_weight(
            name,
            shape=[],
            trainable=False,
            initializer=value,
            aggregation=tf_variables.VariableAggregation.ONLY_FIRST_REPLICA)
    self._hypers_created = True
```

**Describe the expected behavior**

It seems unnecessary to create variables for hyperparameters that are known to be constant.

**Standalone code to reproduce the issue**

This code creates a variable for the learning rate in the graph.

```python
tf.keras.optimizers.SGD(learning_rate=0.01)
```

**Comment by amahendrakar** [comment]
@behzad-a,
On instantiating the optimizer with TF v2.2, the type is a `tensorflow.python.keras.optimizer_v2.gradient_descent.SGD` object.
> This code creates a variable for the learning rate in the graph.

Could you please provide a minimal code to reproduce the issue? Thanks!

**Comment by behzad-a** [comment]
@amahendrakar, I think you misunderstood my question. What I'm referring to are the ops in the TF graph (or the XLA graph for that matter) that are created by using these optimizers in a computation. Keras V2 optimizers, such as `tensorflow.python.keras.optimizer_v2.gradient_descent.SGD`, internally convert their **constant** hyperparameters (learning rate, momentum, etc.) into **non-trainable variables** in the TF graph as opposed pure constants (like tf.constant for example).

**Comment by behzad-a** [comment]
Any updates on this issue? Thanks.

**Comment by rmothukuru** [comment]
@behzad-a,
Can you please share the complete reproducible code or a Colab Gist so that we can take it forward? Thanks!

**Comment by behzad-a** [comment]
Here is a simple example of reproducing this:

```python
import tensorflow as tf
optimizer = tf.keras.optimizers.SGD(learning_rate=0.1)  # opt with a constant hyperparameter (LR)
lr_op = optimizer._get_hyper("learning_rate")  # hyperparameter that's created in the TF graph
print(isinstance(lr_op, tf.Variable))  # Prints True
print(isinstance(lr_op, tf.Tensor))  # Prints False
```

**Comment by rmothukuru** [comment]
@behzad-a,
**`Learning Rate`** is set as **`Variable`** because some **`Optimizers`** change the value of **`Learning Rate`** dynamically during **`Training`**. Example of one such Optimizer is [Adagrad](https://www.tensorflow.org/api_docs/python/tf/keras/optimizers/Adagrad). 
Hope this answers your question. Thanks!

**Comment by behzad-a** [comment]
While that may be true for Adagrad, that's not always the case. For an SGD optimizer, for example, a constant learning rate is guaranteed to stay constant during training. Furthemore, please note that this odd behavior is only observed in V2 optimizers. V1 optimizers, such as the one pasted below, correctly use constants in the TF graph for learning rates that are explicitly specified to be constants by the user.

```python
optimizer = tf.compat.v1.train.GradientDescentOptimizer(learning_rate=0.1)
```

**Comment by behzad-a** [comment]
Any updates on this? Thanks.

**Comment by fchollet** [comment]
The reason is that these parameters are intended to be mutable without rebuilding the graph (graphs are immutable), e.g. `optimizer.learning_rate = 0.1` is a valid use case (independently of the optimizer considered).
